fdbclient/BlobGranuleFiles.cpp

```diff
@@ -378,7 +379,8 @@ struct IndexBlockRef {
 		}
 	}
 
-	void finalize(Optional<BlobGranuleCipherKeysCtx> cipherKeysCtx, Arena& arena) {
+	void finalize(Optional<BlobGranuleCipherKeysCtx> cipherKeysCtx, Arena& arena, EncryptionAtRestMode encryptMode) {
+		ASSERT(!encryptMode.isEncryptionEnabled() || cipherKeysCtx.present());
```

Comment:
We can avoid passing encryptionAtRest mode here, essentially having an assert where the `cipherKeysCtx` gets populated is good enough?

Reply:
I thought of doing this my reason for this change is there is alot of code in between when the cipherKeys get populate to where it is used. This assert makes sure that right before encryption we have the correct state

Reply:
I'm not opposed to doint at the level `cipherKeysCtx` is populated either but i thought this is a bit stricter

Reply:
In some ways it is redundant, `cipherKeyCtx` are populated in two ways: 
1. WriteSnapshot - sets the `Optional` **iff** encryption is enabled.
2. ReadSnapshot - `Optional` is set by reading the metadata from BlobGranuleMetata and gets populated. 

I don't think we should modify all these APIs, however, we should ensure that caller _should_ populate `Optional` in the right way.